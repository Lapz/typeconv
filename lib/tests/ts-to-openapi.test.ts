import { makeConverter, getTypeScriptReader, getOpenApiWriter } from "../"


describe( "ts-to-openapi", ( ) =>
{
	it( "typescript to openapi", async ( ) =>
	{
		const input = `
			type Thing = {
				x: 6
				y: string
			}

			export type Foo = {
				a: string
				b: null
				c: number
				d: boolean
				e: Thing
			}
		`;

		const { convert } = makeConverter(
			getTypeScriptReader( ),
			getOpenApiWriter( {
				format: 'yaml',
				title: 'My API',
				version: 'v1'
			} )
		);

		const { data } = await convert( { data: input } );

		expect( data ).toMatchSnapshot( );
	});
} );
