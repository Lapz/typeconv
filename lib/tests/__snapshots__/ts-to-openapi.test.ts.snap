// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ts-to-openapi typescript to openapi 1`] = `
"openapi: 3.0.0
info:
  title: My API
  version: v1
  x-comment: >-
    Generated by core-types-json-schema
    (https://github.com/grantila/core-types-json-schema) on behalf of typeconv
    (https://github.com/grantila/typeconv)
paths: {}
components:
  schemas:
    Foo:
      properties:
        a:
          type: string
        b:
          nullable: true
        c:
          type: number
        d:
          type: boolean
        e:
          $ref: '#/components/schemas/Thing'
      required:
        - a
        - b
        - c
        - d
        - e
      additionalProperties: false
      type: object
    Thing:
      properties:
        x:
          enum:
            - 6
          type: number
        'y':
          type: string
      required:
        - x
        - 'y'
      additionalProperties: false
      type: object
"
`;

exports[`ts-to-openapi typescript to openapi 2`] = `
"openapi: 3.0.0
info:
  title: My API
  version: v1
  x-comment: >-
    Generated by core-types-json-schema
    (https://github.com/grantila/core-types-json-schema) on behalf of typeconv
    (https://github.com/grantila/typeconv)
paths: {}
components:
  schemas:
    Point:
      properties:
        x:
          title: The distance from the left in mm
          type: number
        'y':
          title: The distance from the top in mm
          type: number
      required:
        - x
        - 'y'
      additionalProperties: false
      type: object
"
`;
